### ISSUE TITLE
[entropy_src/rtl] add a startup phase for health testing with its set of window and threshold regs

### ISSUE THREAD

**Issue by mwbranstad** [lowRISC/opentitan#6085]
The requirements for the startup health tests appear to be different from the continuous health test phases. For this reason, an extra set of threshold registers and a new window registers need to be added. With these new values, the hardware sequencer can choose between the boot phase, the startup phase, and the continuous phase. 

The attached diagram shows the health test phases and what the new hardware requirements are.
[ht_phases.pdf](https://github.com/lowRISC/opentitan/files/6305148/ht_phases.pdf)

**Comment by mwbranstad** [comment]
For entropy meeting discussion: @tjaychen @martin-lueker @cdgori @felixmiller @moidx

**Comment by mwbranstad** [comment]
forget @senelson7 @zi-v

**Comment by mwbranstad** [comment]
Meeting review was that the current thresholds and window register should be re-programmed when transitioning from startup phase to continuous checking phase. This means less registers and documentation updates. However, to make this work, the stop/start feature is needed, and probably a per transaction register lock function.

**Comment by zi-v** [comment]
Some comments regarding health tests with reference to NIST.SP.800-90B.

In '4.4.2 Adaptive Proportion Test' section, the flow and window size is defined as follows:
"The test takes a sample from the noise source, and then counts the number of times that the same value occurs within the next W-1 samples. If that count reaches the cutoff value C, the test declares an error. The window size W is selected based on the alphabet size, and shall be assigned to 1024 if the noise source is binary (that is, the noise source produces only two distinct values) and 512 if the noise source is not binary (that is, the noise source produces more than two distinct values)"

From the above, I gather that:
- We are required to test only one 'bucket' at a time (it is not forbidden to test several buckets in parallel, just need to manage the false-positive rate).
- For our case, the APT window size is strictly defined to 512 samples. It doesn't sound like it is allowed to run any different window size (neither larger nor smaller).

Section '4.3 Requirements for Health Tests' , subsection 4 says:
"The entropy source's startup tests shall run the continuous health tests over at least 1024 consecutive samples. The startup tests may include other tests defined by the developer. The samples subjected to startup testing may be released for operational use after the startup tests have been passed, or may be discarded at any time."

From the combination of the two above quotes, my best guess is that in our case, we should run the ATP twice with regular (continuous test) settings and only after both passed, we can start releasing bits to the entropy consumers.

To sum up, I am not sure a reconfiguration is needed or even allowed. Moreover, during the reconfiguration, we might miss some samples which may be disallowed by itself. I suggest we reiterate with a lab to make sure we do the right thing.

**Comment by mwbranstad** [comment]
[ht_phases2.pdf](https://github.com/lowRISC/opentitan/files/6378704/ht_phases2.pdf)

Added updated drawing based on discussions.
